packages/flutter/lib/src/foundation/_timeline_web.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:js/js.dart';
+
+/// Returns the current timestamp in microseconds from a monotonically
+/// increasing clock.
+///
+/// This is the web implementation, which uses `window.performance.now` as the
+/// source of the timestamp.
+///
+/// See:
+///   * https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+double get performanceTimestamp => 1000 * _performance.now();
```

Comment:
Is there a reason `performanceTimestamp` is a double instead of `1000 * _performance.now().toInt()`?

Reply:
I tried, but since there's no `int64` or `Int64List` on the web, I have to store these as doubles anyway. Otherwise, I'd have to store values in `List<int>` (which on the web is an array of `Number`s, i.e. `double`) and lose some performance predictability. This code relies on the fact that I can allocate small fixed-sized `Float64List` instances for a constant cost.

Reply:
Ah, I was confused by this line:
```dart
static int get now => impl.performanceTimestamp.toInt();
```
But I see that it's being retained as a double elsewhere.